Haptics/app/src/main/java/com/example/android/haptics/samples/ui/views/HomeViewModel.kt

```diff
@@ -0,0 +1,257 @@
+                    hapticFeedbackCategories = listOf(
+                        // List of predefined effects.
+                        HapticFeedbackCategory(
+                            application.getString(R.string.home_screen_predefined_effects),
+                            categoryType = HapticFeedbackCategoryType.PREDEFINED_EFFECTS,
+                            buttons = listOf(
+                                HapticFeedbackButton(
+                                    label = application.getString(R.string.home_screen_tick),
+                                    worksOnUserDevice = isEffectSupported(
+                                        vibrator,
+                                        VibrationEffect.EFFECT_TICK
+                                    ),
+                                    vibrationEffectConstant = VibrationEffect.EFFECT_TICK
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_click),
+                                    isEffectSupported(
+                                        vibrator,
+                                        VibrationEffect.EFFECT_CLICK
+                                    ),
+                                    VibrationEffect.EFFECT_CLICK
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_heavy_click),
+                                    isEffectSupported(vibrator, VibrationEffect.EFFECT_HEAVY_CLICK),
+                                    VibrationEffect.EFFECT_HEAVY_CLICK
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_double_click),
+                                    isEffectSupported(vibrator, VibrationEffect.EFFECT_DOUBLE_CLICK),
+                                    VibrationEffect.EFFECT_DOUBLE_CLICK
+                                ),
+                            )
+                        ),
+                        // List of haptic feedback constants.
+                        HapticFeedbackCategory(
+                            application.getString(R.string.home_screen_feedback_constants),
+                            categoryType = HapticFeedbackCategoryType.HAPTIC_CONSTANTS,
+                            buttons = listOf(
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_heavy_click),
+                                    androidRorLater,
+                                    HapticFeedbackConstants.CONFIRM
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_submit_failure),
```

Comment:
Maybe we can name these confirm and reject, to match the HapticFeedbackConstants API and keep consistency? Might be worth changing the actual string to those names as well.

Reply:
Done